### ISSUE TITLE
QA-pipeline fails to start: Parameter 1 of method configurator in QanaryPipeline required a bean of type QanaryTripleStoreConnector that couldnt be found

### ISSUE THREAD

**Issue by JojoMee** [WDAqua/Qanary#88]
## Description of the bug
After compiling the qanary pipeline from the current `master` branch it fails to start with the following error message (full debug log attached at the bottom):
```
***************************
APPLICATION FAILED TO START
***************************

Description:
Parameter 1 of method configurator in eu.wdaqua.qanary.QanaryPipeline required a bean of type 'eu.wdaqua.qanary.commons.triplestoreconnectors.QanaryTripleStoreConnector' that could not be found.

Action:
Consider defining a bean of type 'eu.wdaqua.qanary.commons.triplestoreconnectors.QanaryTripleStoreConnector' in your configuration.
```

My system configuration:

```
qanary@qanary:/opt/Qanary$ lsb_release -drc
Description:    Ubuntu 22.04 LTS
Release:        22.04
Codename:       jammy

qanary@qanary:/opt/Qanary$ java --version
openjdk 11.0.15 2022-04-19
OpenJDK Runtime Environment (build 11.0.15+10-Ubuntu-0ubuntu0.22.04.1)
OpenJDK 64-Bit Server VM (build 11.0.15+10-Ubuntu-0ubuntu0.22.04.1, mixed mode, sharing)

qanary@qanary:/opt/Qanary$ mvn --version
Apache Maven 3.6.3
Maven home: /usr/share/maven
Java version: 11.0.15, vendor: Private Build, runtime: /usr/lib/jvm/java-11-openjdk-amd64
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux", version: "5.13.19-6-pve", arch: "amd64", family: "unix"
```

A Stardog database is running at `localhost` with a database named `qanary`:
```
qanary@qanary:/opt/Qanary$ docker ps
CONTAINER ID   IMAGE             COMMAND                  CREATED      STATUS      PORTS                                       NAMES
a3bb292927ad   stardog/stardog   "/opt/stardog/bin/st…"   2 days ago   Up 2 days   0.0.0.0:5820->5820/tcp, :::5820->5820/tcp   adoring_gagarin

qanary@qanary:/opt/Qanary$ curl http://admin:admin@localhost:5820/qanary
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix : <http://api.stardog.com/> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix stardog: <tag:stardog:api:> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
```

## Steps to reproduce
Follow the steps from the readme to compile and start the qanary pipeline:
```shell
git clone https://github.com/WDAqua/Qanary
cd Qanary
mvn clean install -Ddockerfile.skip=true
java -jar qanary_pipeline-template/target/qa.pipeline-3.1.0.jar --qanary.triplestore="http://admin:admin@localhost:5820/qanary"
```

This assumes a stardog database is running at `localhost` with the default credentials and a database named `qanary`. I've also changed all loglevels in the `application.properties` to `DEBUG`.

## More details
Full debug log:
[qanary-debug-log.txt](https://github.com/WDAqua/Qanary/files/8593173/qanary-debug-log.txt)

Normal log (loglevel info/warn):
[qanary-normal-log.txt](https://github.com/WDAqua/Qanary/files/8594390/qanary-normal-log.txt)

**Comment by JojoMee** [comment]
I figured out that adding the stardog configuration to `application.properties` fixes this issue:
```properties
# [...]

### a triplestore needs to be configured to store the data of any Qanary question answering process
### since Qanary v3 you might implement your own triplestore connector as a subclass of eu.wdaqua.qanary.commons.triplestoreconnectors.QanaryTripleStoreConnector
### Stardog: if you choose to use the Stardog triplestore, then the following properties are required to create an instance of QanaryTripleStoreConnector
### see the class for additional configuration properties interpreted by Stardog
stardog.url=http://127.0.0.1:5820/qanary
stardog.username=admin
stardog.password=admin
```

Then one can start the qanary pipeline without the `--qanary.triplestore` argument:
```sh
java -jar qanary_pipeline-template/target/qa.pipeline-3.1.0.jar
```

I'm still leaving this open as I'm not sure if there is a bug in the code that loads the configuration or the documentation on how to configure the pipeline properly was not updated in the readme.

**Comment by Wutras** [comment]
@JojoMee Hi, thanks for your detailed description! You've already come up with the solution and I'd like to add to it by pointing to the relevant wiki pages in case anyone faces a similar issue in the future.

This page details the configuration of a Qanary pipeline: [https://github.com/WDAqua/Qanary/wiki/Configuration-Parameters-of-a-Qanary-Pipeline](https://github.com/WDAqua/Qanary/wiki/Configuration-Parameters-of-a-Qanary-Pipeline)
This page illustrates how to set up your own triplestore, pipeline and components using an example: [https://github.com/WDAqua/Qanary/wiki/Qanary-tutorial:-How-to-build-a-trivial-Question-Answering-pipeline](https://github.com/WDAqua/Qanary/wiki/Qanary-tutorial:-How-to-build-a-trivial-Question-Answering-pipeline)

And yeah, the documentation in the README seems to be outdated. In the most recent version of the wiki, the following is written:

> If a Stardog triplestore is used, then the following properties need to be defined:
> 
> - `stardog.url`
> - `stardog.username`
> - `stardog.password`
> 
> Note: You may implement your own triplestore connector that requires different properties.
> 
> Additionally, the property `qanary.triplestore` provides a fallback option if:
> * the Stardog triplestore is running on a different host to the pipeline 
> * you cannot define your own triplestore and no other options are available

I'll go ahead and update the README now, thanks for pointing this out!

**Comment by JojoMee** [comment]
resolved by #90
